core/src/main/kotlin/fr/sncf/osrd/stdcm/infra_exploration/InfraExplorer.kt

```diff
@@ -271,8 +271,11 @@ private class InfraExplorerImpl(
         var nBlocksToSkip = 0
         val pathFragments = arrayListOf<PathFragment>()
         var pathStarted = incrementalPath.pathStarted
+        val addedBlocks = mutableListOf<BlockId>()
 
         for (block in routeBlocks) {
+            addedBlocks.add(block)
+            blocks.add(block)
```

Comment:
Do we want to update the doc and clarify `blocks` may be changed even if false is returned?

Reply:
I went too fast when fixing that and only reverted part of the change, the state is now unchanged when false is returned